inspirehep/modules/workflows/tasks/submission.py

```diff
@@ -97,3 +289,104 @@ def _close_ticket(obj, eng):
                     raise
                 obj.log.warning("Ticket is already resolved.")
     return _close_ticket
+
+
+def halt_record_with_action(action, message):
+    """Halt the record and set an action (with message)."""
+    @wraps(halt_record_with_action)
+    def _halt_record(obj, eng):
+        """Halt the workflow for approval."""
+        eng.halt(action=action,
+                 msg=message)
+    return _halt_record
+
+
+def send_robotupload(url=None,
+                     callback_url="callback/workflows/continue",
+                     mode="insert"):
+    """Get the MARCXML from the model and ship it."""
+    @wraps(send_robotupload)
+    def _send_robotupload(obj, eng):
+        from invenio_workflows.errors import WorkflowError
+        from inspirehep.utils.robotupload import make_robotupload_marcxml
+
+        combined_callback_url = os.path.join(current_app.config["SERVER_NAME"], callback_url)
+        model = eng.workflow_definition.model(obj)
+        sip = model.get_latest_sip()
+        marcxml = sip.package
+        result = make_robotupload_marcxml(
+            url=url,
+            marcxml=marcxml,
+            callback_url=combined_callback_url,
+            mode=mode,
+            nonce=obj.id
```

Comment:
On `production` branch I had to amend this for `authors` and for `suggestion`. Can you check if this has now changed and this is the only place where  `make_robotupload_marcxml` need to be amended?